prime-router/src/main/kotlin/Report.kt

```diff
@@ -855,17 +862,46 @@ class Report : Logging {
                             null
                         }
                     }
-                    it.specimenCollectionMethod = row.getStringOrNull("specimen_collection_method")
-                    it.specimenCollectionSite = row.getStringOrNull("specimen_collection_site")
-                    it.specimenType = row.getStringOrNull("specimen_type")
+                    it.specimenCollectionMethod = row.getStringOrNull("specimen_collection_method_text")
+                    it.specimenCollectionMethodCode = row.getStringOrNull("specimen_collection_method_code")
+                    it.specimenCollectionSite = row.getStringOrNull("specimen_collection_site_text")
+                    it.specimenCollectionSiteCode = row.getStringOrNull("specimen_collection_site_code")
+                    it.specimenType = row.getStringOrNull("specimen_type_name")
+                    it.specimenTypeCode = row.getStringOrNull("specimen_type_code")
+                    it.specimenTypeNormalized = if (it.specimenTypeCode != null) {
+                        metadata.findValueSet("monkeypox/specimen_type")?.toDisplayFromCode(it.specimenTypeCode)
+                    } else {
+                        null
+                    }
                     it.specimenSourceSite = row.getStringOrNull("specimen_source_site_text")
+                    it.specimenSourceSiteCode = row.getStringOrNull("specimen_source_site_code")
 
                     it.siteOfCare = row.getStringOrNull("site_of_care").trimToNull()
                     it.testKitNameId = row.getStringOrNull("test_kit_name_id").trimToNull()
                     it.testPerformedCode = row.getStringOrNull("test_performed_code").trimToNull()
                     it.testPerformed = row.getStringOrNull("test_performed_name").trimToNull()
+                    it.testPerformedNormalized = if (it.testPerformedCode != null) {
+                        metadata.findValueSet("monkeypox/test_code")?.toDisplayFromCode(it.testPerformedCode)
+                    } else {
+                        null
+                    }
+                    it.testPerformedLongName = if (it.testPerformedCode != null) {
+                        metadata.findValueSet("monkeypox/test_long_name")?.toDisplayFromCode(it.testPerformedCode)
+                    } else {
+                        null
+                    }
```

Comment:
Tests can have long names, which are part of the PHINVADS data. Let's trap this too.